### PR TITLE
Setup for rust-vmm-ci-ci

### PR DESCRIPTION
**Pull request by catalindumitru**
In order to be able to run the CI on rust-vmm-ci we needed to
make it a rust crate. Also the path to the coverage config
file needed to be updated in test_coverage.py so that the path
does not depend on the crate; a breadth-first search was used
to guarantee that the config file belongs to the crate that is
being tested.

In the other crates, the tests reside in rust-vmm-ci/intergation_tests.
When we test the CI itself, the tests reside in integration_tests.
Therefore we needed to update the pytest commands to find the tests 
independently of the crate they are run from.

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -172,7 +172,7 @@ steps:
 
   - label: "coverage-x86"
     commands:
-      - pytest rust-vmm-ci/integration_tests/test_coverage.py
+      - find . -type f -name "test_coverage.py" | xargs pytest
```
Comment (andreeaflorescu) [inline .buildkite/pipeline.yml:175]: The changes to the pipeline should be in a separate commit and we should explain in the commit message why are they needed.

File: coverage_config_aarch64.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (andreeaflorescu) [inline coverage_config_aarch64.json:1]: The aarch64 coverage test is not yet running. We can remove this file and add it only when we need it.

File: integration_tests/utils.py [hunk 3]
```diff
@@ -13,6 +13,21 @@ def get_repo_root_path():
     return os.path.dirname(rust_vmm_ci_path)
 
 
+def get_coverage_config_path(platform):
```
Comment (andreeaflorescu) [inline integration_tests/utils.py:16]: Since this is not a utility we expect to use in other tests beside the coverage one, we can add this function in the test_coverage.py file.

File: integration_tests/utils.py [hunk 4]
```diff
@@ -13,6 +13,21 @@ def get_repo_root_path():
...
+    target_file = f"coverage_config_{platform}.json"
+    dirs = [os.getcwd()]
```
Comment (andreeaflorescu) [inline integration_tests/utils.py:18]: Let's add a comment here why we are not using os.walkdir so that in the future people don't try to wrongly optimize it.

File: coverage_config_aarch64.json (hunk 2 above)
Comment (catalindumitru) [reply coverage_config_aarch64.json]: Done.

File: integration_tests/utils.py (hunk 4 above)
Comment (catalindumitru) [reply integration_tests/utils.py]: Done.

File: .buildkite/pipeline.yml (hunk 1 above)
Comment (catalindumitru) [reply .buildkite/pipeline.yml]: Done.

File: integration_tests/utils.py (hunk 3 above)
Comment (catalindumitru) [reply integration_tests/utils.py]: Done.

**Review by andreeaflorescu — APPROVED** [review]

File: coverage_config_x86_64.json [hunk 5]
```diff
@@ -0,0 +1,5 @@
+{
+    "coverage_score": 33.3,
```
Comment (lauralt) [inline coverage_config_x86_64.json:2]: Why 33.3 if we have no tests? I also just ran `find . -type f -name "test_commit_format.py" | xargs pytest` in container and the reported value seems to be 0.

Comment (lauralt) [reply coverage_config_x86_64.json:2]: Hmm, looks like it depends on the Rust version; from 1.49.0 the reported coverage is 33.3 and before -> 0. Anyway it's not related to the changes in this PR, so disregard my comment.

**Review by lauralt — APPROVED** [review]
